hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/protocol/datatransfer/sasl/SaslDataTransferClient.java

```diff
@@ -605,14 +605,14 @@ private IOStreamPair doSaslHandshake(InetAddress addr,
     } catch (IOException ioe) {
       try {
         sendGenericSaslErrorMessage(out, ioe.getMessage());
-      } catch (Exception ioe2) {
+      } catch (Exception e) {
         // If ioe is caused by error response from server, server will close peer connection.
         // So sendGenericSaslErrorMessage might cause IOException due to "Broken pipe".
         // We suppress IOException from sendGenericSaslErrorMessage
         // and always throw `ioe` as top level.
         // `ioe` can be InvalidEncryptionKeyException or InvalidBlockTokenException
         // that indicates refresh key or token and are important for caller.
-        ioe.addSuppressed(ioe2);
+        ioe.addSuppressed(e);
```

Comment:
How about adding debug log for the exception `e`? Reading https://github.com/apache/hadoop/blob/b34dcb5b3a3ca45dbb02393e739166e5ca3116dd/hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/FileChecksumHelper.java#L535-L556, I think the suppressed exception is not logged anywhere.

Reply:
I'm +1 if that is addressed. Thanks!

Reply:
Thank you. nice idea. I've addressed in https://github.com/apache/hadoop/pull/3677/commits/48c75cedcba6eed622bfd3868380f6bf3fa4c1db